src/fr-webauthn/interfaces.ts

```diff
@@ -1,11 +1,86 @@
 import HiddenValueCallback from '../fr-auth/callbacks/hidden-value-callback';
 import MetadataCallback from '../fr-auth/callbacks/metadata-callback';
+import TextOutputCallback from 'fr-auth/callbacks/text-output-callback';
+
+enum AttestationType {
+  Direct = 'direct',
+  Indirect = 'indirect',
+  None = 'none',
+}
+
+interface AuthTreeCallback {
+  _id?: number;
+  input?: NameValuePair[];
+  output: NameValuePair[];
+  type: AuthTreeCallbackType;
+}
+
+enum AuthTreeCallbackType {
```

Comment:
Are you suggesting since this is just used for typing on line 15 above, declare it like this:

```js
enum AuthTreeCallbackType {
  ChoiceCallback,
  HiddenValueCallback,
  NameCallback,
  PasswordCallback,
  TextOutputCallback,
}
```

... and then use `keyof` on line 15 above like this:

```js
  type: keyof AuthTreeCallbackType;
```
Is that what you mean?

Reply:
yes :)